### PR TITLE
Fixed issue with connection._registrations

### PR DESCRIPTION
**Pull request by ambrons**
Previous there was an issue raised here:

> TODO: Don't use _registrations and instead use a public API
> related issue: https://github.com/hapijs/hapi/issues/2777

This was fixed and closed.  I was experiencing issues running `http://localhost:8888/setup` as documented
here:  https://github.com/continuationlabs/hapi-setup/issues/28

Switched to using `connection.registrations` and `connection.registration.attributes`

Fixed tests that were failing due to connection count and server.inject
  Hapi's server.inject requires exactly 1 connection in order to work properly.
  With a Hapi server that has multiple connections you must first selected a labeled
  connection prior to calling inject, otherwise an exception is thrown.

Fixed the response data from plugins to support the new Hapi format with `pkg` nested in
  attributes.

Updated package.json to internalize `lab` usage to the node_modules/lab/bin/lab app, removing
  the require ment that lab be globally installed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,7 +44,7 @@
     "lint": "belly-button -i 'lib/*.js' -i dev-server.js -i webpack.config.js",
     "prepublish": "webpack",
     "start": "node dev-server & npm run webpack",
-    "test": "npm run lint && lab -v -t 100 -a code",
+    "test": "npm run lint && ./node_modules/lab/bin/lab -v -t 100 -a code",
```
Comment (arb) [inline package.json]: You shouldn't need the path like that. `npm` scripts assume that already.

File: test/index.js [hunk 2]
```diff
@@ -243,7 +248,12 @@ describe('hapi-setup Plugin', function () {
           },
           attributes: {
             name: 'bar',
-            version: '1.0.0'
+            version: '1.0.0',
+            pkg: {
+              version: '0.0.0'
+            },
+            multiple: false,
```
Comment (arb) [inline test/index.js:255]: Did you need to add these? I'm pretty sure hapi defaults them if they are missing.

File: test/index.js [hunk 3]
```diff
@@ -243,7 +248,12 @@ describe('hapi-setup Plugin', function () {
           },
           attributes: {
             name: 'bar',
-            version: '1.0.0'
+            version: '1.0.0',
+            pkg: {
+              version: '0.0.0'
```
Comment (arb) [inline test/index.js:253]: Please copy the same version this was already using.

File: test/index.js (hunk 2 above)
Comment (ambrons) [inline test/index.js:255]: I had to add them to make the test pass.  So perhaps the issue is how the test works.  Again I'm making assumptions on how this should have worked considered all tests where failing.

File: test/index.js (hunk 3 above)
Comment (ambrons) [inline test/index.js:253]: I'm assuming you're referring to the issue with 

```
pkg: {
    version: '0.0.0'
}
```

That's actually what is returned via Hapi in this case.  I wasn't adding it to not match.  So perhaps this is a symptom of a upstream problem.

Comment (arb) [inline test/index.js:253]: Actually I'm really surprised you needed to change this at all? Can you please explain why this change was necessary? The plugin interface hasn't changed in like 5 versions of hapi.

Comment (ambrons) [inline test/index.js:253]: I see what you mean, the data is the same in both cases, however expect seems to think it's different.

```
Failed tests:

  4) hapi-setup Plugin returns plugin information:

      actual expected

      {
        "name": "foo",
        "version": "0.0.0",
        "multiple": false,
        "options": {},
        "attributes": {
          "name": "foo",
          "pkg":
        {
            "version": "0.0.0"
          },
          "multiple": false,
          "connections": true
        }
      }

      Expected { name: 'foo',
  version: '0.0.0',
  multiple: false,
  options: {},
  attributes: 
   { name: 'foo',
     pkg: { version: '0.0.0' },
     multiple: false,
     connections: true } } to equal specified value

      at /Users/abrons/devel/adac/hapi-setup/test/index.js:228:37
```

Looks like perhaps a formatting issue, but that doesn't really make sense.

**Comment by arb** [comment]
Hey @ambrons thanks for the PR. I've incorporated some of your changes into my PR that should get this back to ship-shape working with hapi. There were a few subtle changes that impacted how information was presenting via `connection.registrations`.
